projects/plugins/boost/app/modules/cache/file_cache.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+require_once __DIR__ . '/cache.php';
+
+class Boost_File_Cache extends Boost_Cache {
+	private $path           = false;
+	private $cache_filename = false;
+
+	private function calculate_path( $request_uri ) {
+		$key  = md5( $request_uri );
+		$path = WP_CONTENT_DIR . DIRECTORY_SEPARATOR . 'boost-cache' . DIRECTORY_SEPARATOR . 'cache' . DIRECTORY_SEPARATOR;
+		for ( $i = 0; $i <= 5; $i++ ) {
+			$path .= substr( $key, $i, 1 ) . DIRECTORY_SEPARATOR;
+		}
+
+		return $path;
+	}
```

Comment:
I notice that when it creates a directory for cache files, it only takes the request_uri into account, and MD5s it. It will generate a different MD5 hash to the full cache filename which will take the GET params and COOKIE into account. 

What's the benefit of generating a different MD5 hash for the directory structure?

Reply:
It allows us to do admin work on URLs more easily. If a page is edited, we can delete all the cached files for that page by getting the path to it. All the visitors who ever visited that page will have cached pages in that single directory.